content/backend/typescript-apollo/9-deployment.md

```diff
@@ -0,0 +1,692 @@
+
+![Heroku dashboard](https://i.imgur.com/VVXSmkV.png)
+
+Heroku has different kinds of add-ons that can be used to add features or extend your app. You will now use the [heroku-postgres](https://www.heroku.com/postgres) add-on to create a database for your app. 
+
+<Instruction>
+
+Create a PostgreSQL database instance for your Heroku app with the following command: 
+
+```bash(path=".../hackernews-typescript/")
+heroku addons:create heroku-postgresql:hobby-dev
+```
+
+</Instruction>
+
+
+You should see an output *similar* to the following: 
+
+```shell(nocopy)
+Creating heroku-postgresql:hobby-dev on ⬢ app-name-placeholder... free
+Database has been created and is available
+ ! This database is empty. If upgrading, you can transfer
+ ! data from another database with pg:copy
+Created postgresql-cubic-54990 as DATABASE_URL
+```
+
+This command also creates a new environment variable called `DATABASE_URL` in the Heroku app, which contains the connection string to the PostgreSQL database. This is very convenient because when you deploy the API, Prisma will have access to the production database automatically, without any additional configuration on your end. The connection string uses the same format as Prisma too! 
+
+> **Note:** You can check your production database connection string with this command: `heroku config:get DATABASE_URL`. 
+
+Now that we have set up the necessary resources on Heroku, it's time to set up a deployment pipeline that will ensure continuous deployment. 
+
+
+### Implementing continuous deployment using GitHub Actions
+
+GitHub Actions is an automation platform that can be used for continuous integration (CI) and continuous deployment (CD). It has an API for orchestrating workflows based on events in GitHub and can be used to build, test, and deploy your code directly from GitHub.
+
+To configure GitHub Actions, you define _workflows_ using the "yaml" file format. These workflows are stored in the `.github/workflows` directory of your project. You can configure workflows to run on different repository events, e.g., when a commit is pushed to the repository or when a pull request is created.
+
+Each workflow can contain one or more _jobs_ which are dedicated tasks, which run in an isolated environment (known as a "runner"). A single job can also have one or more steps. In this project, you will create a single workflow with a job called `deploy`. 
+
+<Instruction>
+
+Create a `.github/workflows` directory and a new workflow inside it. 
+
+```bash(path=".../hackernews-typescript/")
+mkdir .github
+mkdir .github/workflows
+touch .github/workflows/deployment.yml
+```
+
+</Instruction>
+
+<Instruction>
+
+Define the `deploy` job inside the newly created workflow:
+
+```yaml{}(path=".../hackernews-typescript/.github/workflows/continuous-integration.yml")
+name: deploy-hackernews-app-heroku                       # 1
+
+on:
+  push:
+    branches:                                            # 2
+      - master
+      - main
```

Comment:
Perhaps it's better to explicitly keep both master and main? Most of the tutorials on the website still use the master branch, so it might be frustrating to new users if their code does not work as expected if they just directly copy-paste this code snippet.